### DISCUSSION TITLE
US Department of Defense policy on modules

### DISCUSSION THREAD

**Discussion by potatoqualitee** [PowerShell/PowerShell#19303] [Q&A]
Is the PS Team aware, perhaps through experience or through a Microsoft PFE, of the DoD or DISA's position on module usage and in particular, the PowerShell Gallery?

With Linux distros, there's no additional scrutiny on `apt get`; it's a blanket approval on the specific distro and version. I did find specific approval on PowerShell 5.1 and 7.x and I'm wondering if that also applies to modules that run on these versions.

**Comment by doctordns** [comment]
Got a reference to this policy?

**Comment by SeeminglyScience** [comment]
I'm not aware of any DoD SRG/STIGs that outright reference the gallery directly. I'd need to do some research to say that conclusively, but a quick glance does not show one.

That said, the somewhat [recentish executive order](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) on cybersecurity does lay out some general rules regarding dependency management in general. It again does not reference the gallery directly, but it is relevant in some of the directions it gives.

**Reply by jborean93** [reply]
Even if there is no official policy, most places I’ve seen block internet access outright on servers. Typically they set up their own repository that contains the packages they have approved and can be used internally. The policy for approval depends on the org itself.

**Reply by potatoqualitee** [reply]
Thank you both, that's basically what I found too. It's per org with some reciprocity sometimes.
